akka-http2-support/src/test/scala/akka/http/impl/engine/http2/HPackSpecExamples.scala

```diff
@@ -192,7 +193,17 @@ object HPackSpecExamples {
         headers.Date.parseFromValueString("Mon, 21 Oct 2013 20:13:22 GMT").right.get,
         headers.Location("https://www.example.com"),
         headers.`Content-Encoding`(HttpEncodings.gzip),
-        // `Set-Cookie` modeled header doesn't support extra params like "version=1"
+        // The RFC example spells 'max-age' in lowercase, while the modeled `Set-Cookie` header renders it uppercase.
+        // Rendering it uppercase is consistent with https://tools.ietf.org/html/rfc6265#section-5.2.2 .
+        // https://tools.ietf.org/html/rfc7540#section-8.1.2 specifies HTTP/2 header fields ('Set-Cookie') must be
+        // converted to lowercase, but AFAICS has no such provision for header parameters.
```

Comment:
Ugh, yeah, for some reason the RFC 6265 uses mixed-case parameter names and later specified that you need to match them case-insensitively...

https://tools.ietf.org/html/rfc6265#section-4.1.1

And yes, the HTTP/1.1 and HTTP/2 specs treat header values opaquely and refer to the particular spec for parsing details.

Reply:
So that's indeed fine here, that we need to use a RawHeader in cases where we want a particular rendering on the wire (which is not related to HTTP/2 itself after all).

Thanks for adding the comment!